packages/subscription-manager/src/components/UserSettings/UserSettings.tsx

```diff
@@ -32,11 +39,20 @@ type NoticeProps = {
 };
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars -- until we start using any of these props
-const UserSettings = ( { value = {}, loading }: UserSettingsProps ) => {
+const UserSettings = ( { value = {}, loading, userId }: UserSettingsProps ) => {
 	const [ formState, setFormState ] = useState< SubscriptionUserSettings >( value );
 	const { mutate, isLoading, isSuccess, error } =
 		Reader.useSubscriptionManagerUserSettingsMutation();
 	const [ notice, setNotice ] = useState< NoticeProps | null >( null );
+	const subscriberEmailAddress = useSubscriberEmailAddress();
+
+	useEffect( () => {
+		if ( userId ) {
+			page.redirect( '/read' );
+		} else if ( ! subscriberEmailAddress ) {
+			page.redirect( '/email-subscriptions' );
```

Comment:
Looks like the page.redirect is just performed after the end of the function, with that the UserSettings component is rendered before the redirect. I'd suggest to do an early return with an empty fragment to avoid the first render. 

```
	if ( userId ) {
		page.redirect( '/read' );
		return <></>;
	} else if ( ! subscriberEmailAddress ) {
		page.redirect( '/email-subscriptions' );
		return <></>;
	}
```